projects/plugins/jetpack/extensions/blocks/donations/edit.js

```diff
@@ -1,102 +1,187 @@
+	'one-time': ONE_TIME_DONATION,
+	'1 month': MONTHLY_DONATION,
+	'1 year': ANNUAL_DONATION,
+};
+
+const DonationTabButton = ( { label, id, tabIndex, isActive, onActivateTab } ) => {
+	return (
+		<div
+			role="button"
+			tabIndex={ tabIndex }
+			className={ classNames( 'donations__nav-item', {
+				'is-active': isActive,
+			} ) }
+			onClick={ () => onActivateTab( id ) }
+			onKeyDown={ () => onActivateTab( id ) }
+			key={ `jetpack-donations-nav-item-${ tabIndex }` }
+		>
+			{ label }
+		</div>
+	);
+};
 
 const Edit = props => {
-	const { attributes, className, setAttributes } = props;
-	const { currency } = attributes;
+	const { attributes, className } = props;
+	const { currency, annualDonation, monthlyDonation, showCustomAmount } = attributes;
 
 	const [ loadingError, setLoadingError ] = useState( '' );
 	const [ shouldUpgrade, setShouldUpgrade ] = useState( false );
 	const [ stripeConnectUrl, setStripeConnectUrl ] = useState( false );
-	const [ products, setProducts ] = useState( [] );
+	const [ products, setProducts ] = useState( defaultProducts );
+	const [ activeTab, setActiveTab ] = useState( DEFAULT_TAB );
 
-	const post = useSelect( select => select( 'core/editor' ).getCurrentPost(), [] );
-	useEffect( () => {
-		setAttributes( { fallbackLinkUrl: post.link } );
-	}, [ post.link, setAttributes ] );
+	const fallbackLinkUrl = useSelect( select => select( 'core/editor' ).getCurrentPost(), [] ).link;
+	const postId = useSelect( select => select( 'core/editor' ).getCurrentPostId(), [] );
 
-	const apiError = message => {
-		setLoadingError( message );
+	const resetActiveTab = ( controlTab, value ) => {
+		if ( value === false && activeTab === controlTab ) {
+			setActiveTab( DEFAULT_TAB );
+		}
 	};
 
 	const filterProducts = productList =>
 		productList.reduce( ( filteredProducts, { id, currency: productCurrency, type, interval } ) => {
 			if ( productCurrency === currency && type === 'donation' ) {
-				filteredProducts[ interval ] = id;
+				const mappedInterval = intervalMap[ interval ];
+				if ( mappedInterval ) {
+					filteredProducts[ mappedInterval ] = id;
+				}
 			}
 			return filteredProducts;
 		}, {} );
 
 	const hasRequiredProducts = productIdsPerInterval => {
 		const intervals = Object.keys( productIdsPerInterval );
-
-		return (
-			intervals.includes( 'one-time' ) &&
-			intervals.includes( '1 month' ) &&
-			intervals.includes( '1 year' )
-		);
+		return Object.keys( defaultProducts ).every( required => intervals.includes( required ) );
 	};
 
 	const mapStatusToState = result => {
-		if ( ( ! result && typeof result !== 'object' ) || result.errors ) {
-			setLoadingError( __( 'Could not load data from WordPress.com.', 'jetpack' ) );
-			return;
-		}
 		setShouldUpgrade( result.should_upgrade_to_access_memberships );
 		setStripeConnectUrl( result.connect_url );
 
+		if ( result.should_upgrade_to_access_memberships || result.connect_url ) {
+			setProducts( defaultProducts );
+			return;
+		}
+
```

Comment:
Moved over here so that we don't perform additional unnecessary operations when we are just going to return.